src/domain/context/fileTree/filter/byFileSearch.ts

```diff
@@ -0,0 +1,40 @@
+import {SchemaDir} from '../../../states/objects/fileTree/SchemaDir';
+import {SchemaTreeItem} from '../../../states/objects/fileTree/SchemaTreeItem';
+
+export function byFileSearch(search: string): (item: SchemaTreeItem) => boolean {
+    if (!search) {
+        return () => true;
+    }
+
+    search = search.toLowerCase();
+
+    return (item: SchemaTreeItem) => {
+        if(item instanceof SchemaDir)
+        {
+            return checkDir(item, search);
+        }
+
+        return item.basename.toLowerCase().includes(search);
+    };
+}
+
+function checkDir(dir: SchemaDir, search: string) : boolean
+{
+    let children = dir.children;
+    for(let i = children.length - 1; i >= 0; i--)
```

Comment:
The order is unimportant, but helps to visualize the process in my opinion since dirs are at the beginning of the array, followed by the files of the current dir.
I can change this to a for each loop though if desired

Reply:
Then leave it like this 👍🏻